### PR TITLE
eigen patch needed for HIP header changes

### PR DESCRIPTION
**Pull request by jeffdaily**
This patch is needed to fix the ROCm TensorFlow build for changes currently in HIP master.

### CODE REVIEW

File: third_party/eigen3/gpu_packet_math.patch [hunk 1]
```diff
@@ -39,4 +39,127 @@
...
\ No newline at end of file
+ };
+--- a/unsupported/Eigen/SpecialFunctions
++++ b/unsupported/Eigen/SpecialFunctions
+@@ -48,6 +48,9 @@
+ }
+
+ #include "src/SpecialFunctions/SpecialFunctionsImpl.h"
++#if defined(EIGEN_HIPCC)
++#include "src/SpecialFunctions/hip_vector_compatibility.h"
```
Comment (AlexVlx) [inline third_party/eigen3/gpu_packet_math.patch]: I suspect we should chance this to UpperCamelCase, to adhere to the coding standards i.e. it should be HipVectorCompatibility.h

**Comment by sunway513** [comment]
jenkins: retest this please

**Comment by sunway513** [comment]
@jeffdaily , I've modified the TF1.14 nightly CI to use your branch, let's watch for results for the following CI check:
http://ml-ci.amd.com:21096/job/tensorflow-upstream-rel1.14-nightly/220/

**Comment by sunway513** [comment]
CI passed, will merge the PR.

**Pull request merged**